src/test/groovy/org/prebid/server/functional/tests/pg/PgDealsOnlySpec.groovy

```diff
@@ -120,6 +120,37 @@ class PgDealsOnlySpec extends BasePgSpec {
         assert bidErrors[0].message == "Bid \"$bidResponseId\" missing required field 'dealid'"
     }
 
+    def "PBS should add dealsonly flag when it is not specified and pgdealsonly flag is set to true"() {
+        given: "Bid request with set pgdealsonly flag, dealsonly is not specified"
+        def bidRequest = BidRequest.defaultBidRequest
+        bidRequest.imp.first().ext.prebid.bidder.generic = new Generic(pgDealsOnly: true)
+        def initialBidderRequestCount = bidder.requestCount
+
+        and: "No line items response"
+        generalPlanner.initPlansResponse(PlansResponse.getDefaultPlansResponse(bidRequest.site.publisher.id))
+
+        and: "Bid response with missing 'dealid' field to check dealsonly flag was added"
+        def bidResponse = BidResponse.getDefaultBidResponse(bidRequest)
+        bidResponse.seatbid[0].bid[0].dealid = null
+        bidder.setResponse(bidRequest.id, bidResponse)
+        def bidResponseId = bidResponse.seatbid[0].bid[0].id
+
+        and: "Line items are fetched by PBS"
+        updateLineItemsAndWait()
+
+        when: "Auction is happened"
+        def auctionResponse = pgPbsService.sendAuctionRequest(bidRequest)
+
+        then: "Bidder was requested"
+        assert initialBidderRequestCount + 1 == bidder.requestCount
+
+        and: "PBS returns an error of missing 'dealid' field in bid which means PBS added dealsonly flag"
```

Comment:
You should also be able to check resolved request in the debug section of response.